pkg/utils/kubernetes/bootstrap_token.go

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) 2018 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package kubernetes
+
+import (
+	"context"
+	"time"
+
+	"github.com/gardener/gardener/pkg/utils"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	bootstraptokenapi "k8s.io/cluster-bootstrap/token/api"
+	bootstraptokenutil "k8s.io/cluster-bootstrap/token/util"
+)
+
+// ComputeBootstrapToken computes and creates a new bootstrap token, and returns it.
+func ComputeBootstrapToken(ctx context.Context, c client.Client, tokenID, description string, validity time.Duration) (secret *corev1.Secret, err error) {
+	secret = &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      bootstraptokenutil.BootstrapTokenSecretName(tokenID),
+			Namespace: metav1.NamespaceSystem,
+		},
+	}
+
+	err = c.Get(ctx, Key(secret.Namespace, secret.Name), secret)
+	if apierrors.IsNotFound(err) {
```

Comment:
Reverse the if condition.
```suggestion
	if !apierrors.IsNotFound(err) {
	  return secret,err
	}
```